### PR TITLE
added the method supportedFiltersAndFields()

### PR DESCRIPTION
**Pull request by jwbargsten**
For newcomers (like me) it is confusing which filter corresponds to
which field. I added a function to give an overview in form of a
data.frame.

Greetings from the CSAMA 2017. I hope it is useful.

### CODE REVIEW

**Comment by jorainer** [comment]
Thanks @jwbargsten . I think that's a useful addition (after some modifications). @mtmorgan any comments?

**Comment by mtmorgan** [comment]
Sounds like a good idea to me; maybe remove suppportedFilters(), or change the output so that the user isn't overwhelmed with too many similar choices?

**Comment by jorainer** [comment]
OK, then I'll review the pull request.

File: R/AnnotationFilter.R [hunk 1]
```diff
@@ -373,6 +376,20 @@ NULL
...
+  class <- gsub("([[:upper:]])", "_\\1", class, perl=TRUE)
+  class <- sub("^_","", class)
+  class <- tolower(class)
+  class
+}
+
+.supportedFiltersAndFields <- function() {
+  filters <- supportedFilters()
+  df <- data.frame(filter=filters, field=.classToField(filters))
+  df
```
Comment (jorainer) [inline R/AnnotationFilter.R]: One thing: the mapping of `GRangesFilter` to a field `g_ranges` will not work - I suggest to list an `NA` for `GRangesFilter`.

**Review by jorainer — COMMENTED** [review]
1) replace the output of `supportedFilters` with the output of your function.
2) Add an unit test.
3) Ensure that `R CMD check` completes without errors or warnings.

**Review by jorainer — APPROVED** [review]
Nice job @jwbargsten . Please send me your email, so I can add you to the contributors.

**Comment by jwbargsten** [comment]
Thanks. My mail is jw@bargsten.org

**Pull request merged**